src/coreclr/vm/method.hpp

```diff
@@ -159,6 +158,12 @@ enum MethodDescFlags
     mdfIsIntrinsic                      = 0x8000  // Jit may expand method as an intrinsic
 };
 
+struct MethodDescCodeData
+{
+    PTR_MethodDescVersioningState MDState;
```

Comment:
Can MethodDesc point to `PTR_MethodDescVersioningState` directly? I am not able to see the purpose of this extra indirection.

Reply:
For now, yes. I've been working with @davidwrighton on this and he has plans to add fields to this for another scenario. It doesn't really hurt for him to change it back, so I can change it to that if you'd prefer.

Reply:
What are those extra fields going to be for? It would be nice to either add comment about the planned additions here; or omit the extra indirection in this PR.